### PR TITLE
feat: Support unusual dependencies in generated libraries for namespace aliasing.

### PR DESCRIPTION
**Pull request by amanda-tarafa**

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
I've regenerated all APIs locally with this change in and there are no related changes.

File: Google.Api.Generator.Utils/SourceFileContext.cs [hunk 1]
```diff
@@ -247,28 +251,40 @@ private class TypeAliaser : CSharpSyntaxRewriter
                     // from other types. These types may not be aliased if used in the context of their own namespace. Type -> namespaces.
                     Dictionary<string, HashSet<string>> nonCollidingAliasedTypes = new Dictionary<string, HashSet<string>>();
 
-                    // Let's alias namespaces until we have no collisions.
+                    // Let's alias namespaces that we are force to alias
+                    foreach(string ns in forceAliases.Where(fns => mustImportNs.Contains(fns)))
```
Comment (jskeet) [inline Google.Api.Generator.Utils/SourceFileContext.cs:255]: Nit: space before `(`

File: Google.Api.Generator.Utils/SourceFileContext.cs [hunk 2]
```diff
@@ -198,7 +200,8 @@ private class TypeAliaser : CSharpSyntaxRewriter
                     IReadOnlyCollection<Typ> seenTypes,
                     IEnumerable<Typ> packageTyps,
                     IReadOnlyDictionary<string, string> wellKnownNamespaceAliases,
-                    IReadOnlyCollection<Regex> avoidAliasingNamespaceRegex)
+                    IReadOnlyCollection<Regex> avoidAliasingNamespaceRegex,
+                    IReadOnlyCollection<string> forceAliases)
```
Comment (jskeet) [inline Google.Api.Generator.Utils/SourceFileContext.cs:204]: Possibly rename to `forcedAliases`? I'd expect `forceAliases` to be Boolean.

File: Google.Api.Generator/CodeGenerator.cs [hunk 3]
```diff
@@ -75,24 +75,34 @@ internal static class CodeGenerator
...
         };
 
+        /// <summary>
+        /// For unaliased source file context, forces the alias on namespaces on listed on value
+        /// when generating the package in key.
+        /// This is useful for when a package depends on another (not mixin) and we know there are name collisions.
+        /// </summary>
+        private static readonly IReadOnlyDictionary<string, IReadOnlyCollection<string>> ForceAliasForLibrary = new Dictionary<string, IReadOnlyCollection<string>>
+        {
+            { "Google.Cloud.Kms.Inventory.V1", new List<string> { "Google.Cloud.Kms.V1" } }
```
Comment (jskeet) [inline Google.Api.Generator/CodeGenerator.cs:96]: Presumably we *could* make this part of configuration eventually.

File: Google.Api.Generator/CodeGenerator.cs [hunk 4]
```diff
@@ -38,7 +38,7 @@ internal static class CodeGenerator
         /// <summary>
         /// Extension registry with everything we need in.
         /// </summary>
-        private static readonly ExtensionRegistry s_registry = new ExtensionRegistry
+        private static readonly ExtensionRegistry Registry = new ExtensionRegistry
```
Comment (jskeet) [inline Google.Api.Generator/CodeGenerator.cs:41]: Not sure whether the changes in the first commit are really worth it, to be honest - we still have other static readonly fields names `s_xyz`. Don't really mind either way.

**Review by jskeet — APPROVED** [review]
Generally fine, just a few nits and a clarification.

File: Google.Api.Generator.Utils/SourceFileContext.cs (hunk 2 above)
Comment (amanda-tarafa) [reply Google.Api.Generator.Utils/SourceFileContext.cs:204]: Yep, done here and everywhere.

File: Google.Api.Generator.Utils/SourceFileContext.cs (hunk 1 above)
Comment (amanda-tarafa) [reply Google.Api.Generator.Utils/SourceFileContext.cs:255]: Done

File: Google.Api.Generator/CodeGenerator.cs (hunk 4 above)
Comment (amanda-tarafa) [reply Google.Api.Generator/CodeGenerator.cs:41]: It was mostly to standardize it on this particular file, where we already had a mix. I'll leave the changes in and we can change the others progressively, or not.

File: Google.Api.Generator/CodeGenerator.cs (hunk 3 above)
Comment (amanda-tarafa) [reply Google.Api.Generator/CodeGenerator.cs:96]: Yes, +100 to that.

**Review by amanda-tarafa — COMMENTED** [review]
All comments addressed, will push shortly and merge on green, thanks!

**Review by jskeet — APPROVED** [review]

**Pull request merged**